architecture-decision-records/0012-keycloak-email-provider.md

```diff
@@ -0,0 +1,53 @@
+# 12. Keycloak Email Provider
+
+**Date:** 2020-11-16
+
+## Context
+
+Set up Keycloak email provider to be able to send transferring bodies emails for updating passwords, and other Keycloak based user actions.
+
+## Decision
+
+Decided to go with GovUK Notify as the email provider because:
+* Re-using existing UK Government service.
+* Ability to leverage GovUK branding to create specific TDR branding for emails.
+* Free service.
+* 24 hour support.
+* NCSC compliant service.
+* Relatively easy to integrate into Keycloak.
+* Supports SMS. Might make things easier if there is a need to implement MFA via text messages.
+
+### Disadvantages to GovUK Notify
+
+* Logging not integrated with TDR application logging on AWS.
+* Another account to manage for TDR.
+
+### Documentation
+
+See here for further information on GovUK Notify: https://www.notifications.service.gov.uk
+
+## Other Options Considered
+
+### AWS Simple Email Service (SES)
+
+#### Advantages
+
+* Integration with other TDR logging events on AWS
```

Comment:
Also, it would be simpler to integrate SES with our existing infrastructure-as-code Terraform scripts.